### PR TITLE
set mode and permissions from PVC annotation (nfs.io/createMode,nfs/createUID,nfs/createGID)

### PR DESCRIPTION
**Pull request by lorenzofaresin**
**What this PR does / why we need it:**
Needed for https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/issues/41

**Which issue(s) this PR fixes (optional, using fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when the PR gets merged):**
Needed for https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/issues/41

### CODE REVIEW

**Comment by lorenzofaresin** [comment]
Please verify again my CLA signature

**Comment by lorenzofaresin** [comment]
Please verify again my CLA signature

File: cmd/nfs-subdir-external-provisioner/provisioner.go [hunk 1]
```diff
@@ -110,11 +110,33 @@ func (p *nfsProvisioner) Provision(ctx context.Context, options controller.Provi
...
-	glog.V(4).Infof("creating path %s", fullPath)
-	if err := os.MkdirAll(fullPath, 0777); err != nil {
+	createMode := os.FileMode(0777)
+	annotationCreateMode, exists := metadata.annotations["nfs.io/createMode"]
+    if exists {
+        annotationCreateModeUInt, _ := strconv.ParseUint(annotationCreateMode, 8, 32)
+        createMode = os.FileMode(annotationCreateModeUInt)
+    }
+
+    createUID := "0"
```
Comment (yonatankahana) [inline cmd/nfs-subdir-external-provisioner/provisioner.go:120]: root:root is really the default everywhere?

File: cmd/nfs-subdir-external-provisioner/provisioner.go [hunk 2]
```diff
@@ -110,11 +110,33 @@ func (p *nfsProvisioner) Provision(ctx context.Context, options controller.Provi
 		}
 	}
 
-	glog.V(4).Infof("creating path %s", fullPath)
-	if err := os.MkdirAll(fullPath, 0777); err != nil {
+	createMode := os.FileMode(0777)
+	annotationCreateMode, exists := metadata.annotations["nfs.io/createMode"]
+    if exists {
```
[the remark this answers is not in the document]
Comment (lorenzofaresin) [reply cmd/nfs-subdir-external-provisioner/provisioner.go]: fixed

File: cmd/nfs-subdir-external-provisioner/provisioner.go [hunk 3]
```diff
@@ -110,11 +110,33 @@ func (p *nfsProvisioner) Provision(ctx context.Context, options controller.Provi
...
+        annotationCreateModeUInt, _ := strconv.ParseUint(annotationCreateMode, 8, 32)
```
[the remark this answers is not in the document]
Comment (lorenzofaresin) [reply cmd/nfs-subdir-external-provisioner/provisioner.go]: fixed

File: cmd/nfs-subdir-external-provisioner/provisioner.go (hunk 1 above)
Comment (lorenzofaresin) [reply cmd/nfs-subdir-external-provisioner/provisioner.go]: it looks like the previous pull request behavior

**Comment by theXodus** [comment]
Is there any movement on this getting merged? Been a couple of months and there does not seem to be an issue.

**Comment by lorenzofaresin** [comment]
Please review

**Comment by yonatankahana** [comment]
i prefer to have at least  one more review on this kind of changes
/cc @kmova 
/cc @wongma7

**Comment by flo-mic** [comment]
How ist the status of this PR? Would like to see this merged as i have currently permissio issues with nextcloud du to the 777 permission flag

**Comment by terrificsoysauce** [comment]
Can someone reopen this PR?

**Comment by nogweii** [comment]
This might be too much work, but could these options be set as parameters to the storage class instead? And then utilize the same templating support that `pathPattern`.

Maybe those parameters can be set with sane defaults, but this would allow flexibility for an cluster admin to ensure that all directories provisioned are a specific mode, or owned by a particular UID/GID, etc.

**Comment by lorenzofaresin** [comment]
@kmova @wongma7 @ashishranjan738 @yonatankahana please review

**Comment by nogweii** [comment]
I think this is still useful, could someone reopen this and review?

**Comment by mahyard** [comment]
Hey everyone,

What's prevented this PR from being merged? @yonatankahana 
If the only obstacle is a lack of reviewers, could you please consider merging it? The proposed change seems valuable and doesn’t appear to introduce critical problems. If there are other concerns or if my suggestion isn’t acceptable, I’d appreciate any insights or alternative solutions.

Thanks! 😊👍

**Pull request closed without merging**